src/zenml/integrations/skypilot/orchestrators/skypilot_base_vm_orchestrator.py

```diff
@@ -271,29 +272,33 @@ def prepare_or_run_pipeline(
                 f"sudo docker login --username $DOCKER_USERNAME --password "
                 f"$DOCKER_PASSWORD {stack.container_registry.config.uri}"
             )
-            task_envs = {
-                "DOCKER_USERNAME": docker_username,
-                "DOCKER_PASSWORD": docker_password,
-            }
+            task_envs["DOCKER_USERNAME"] = docker_username
+            task_envs["DOCKER_PASSWORD"] = docker_password
         else:
             setup = None
-            task_envs = None
 
         # Run the entire pipeline
 
         # Set the service connector AWS profile ENV variable
         self.prepare_environment_variable(set=True)
 
         try:
+            if isinstance(self.cloud, sky.clouds.Kubernetes):
+                run_command = f"$VIRTUAL_ENV{entrypoint_str} {arguments_str}"
```

Comment:
I think this doesn't work right, `$VIRTUAL_ENV` is for example `/opt/venv` and the python is in `/opt/venv/bin/python`. So this needs the `bin` directory to work, and then also a fallback in case `$VIRTUAL_ENV` is not set.

And we need to make sure this is actually working

Reply:
yes this is important @safoinme